nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java

```diff
@@ -213,6 +260,8 @@ private void createHttpServerFromService(final ProcessContext context) throws Ex
         final Double maxBytesPerSecond = context.getProperty(MAX_DATA_RATE).asDataSize(DataUnit.B);
         final StreamThrottler streamThrottler = (maxBytesPerSecond == null) ? null : new LeakyBucketStreamThrottler(maxBytesPerSecond.intValue());
         final int returnCode = context.getProperty(RETURN_CODE).asInteger();
+        final String authenticationUsername = trimToEmpty(context.getProperty(PROP_BASIC_AUTH_USERNAME).evaluateAttributeExpressions().getValue());
+        final String authenticationPassword = trimToEmpty(context.getProperty(PROP_BASIC_AUTH_PASSWORD).evaluateAttributeExpressions().getValue());
```

Comment:
Should there be a custom validate that ensures that when username is set that password is also set?

Also, should it be a requirement that an SSLContext is provided when using basic auth? 

If not, then it seems very insecure for the external client to be submitting the basic auth credentials over plain http.

Reply:
trimToEmpty is taking take of remplacing null by ""

Regarding the SSLcontext, you are partially right as the authentication should be allowed in both mode (because some devices do not allow HTTPS sometimes for performance reasons)

Reply:
I'll force use of HTTPS for basic auth. That's probably safer.

Reply:
@bbende Changes done and only activated with SSL is enabled.